### PR TITLE
Put noisy thrift logs behind a flag

### PR DESCRIPTION
**Pull request by Breakwell**
I've found using there's a log of noise around the extension code with verbose on:

```
I0214 19:32:04.267999  4932 impl_thrift.cpp:346] Thrift: TPipe ::GetOverlappedResult errored GLE=: errno = 109
I0214 19:32:04.268999  4932 impl_thrift.cpp:346] Thrift: TConnectedClient died: TPipe: GetOverlappedResult failed
I0214 19:32:04.526999  5004 impl_thrift.cpp:346] Thrift: TPipe ::GetOverlappedResult errored GLE=: errno = 109
I0214 19:32:04.582000  5004 impl_thrift.cpp:346] Thrift: TConnectedClient died: TPipe: GetOverlappedResult failed
I0214 19:32:07.620999  5596 impl_thrift.cpp:346] Thrift: TPipe ::GetOverlappedResult errored GLE=: errno = 109
I0214 19:32:07.620999  5596 impl_thrift.cpp:346] Thrift: TConnectedClient died: TPipe: GetOverlappedResult failed
I0214 19:32:07.734999  5608 impl_thrift.cpp:346] Thrift: TPipe ::GetOverlappedResult errored GLE=: errno = 109
I0214 19:32:07.735999  5608 impl_thrift.cpp:346] Thrift: TConnectedClient died: TPipe: GetOverlappedResult failed
I0214 19:32:11.105000  6068 impl_thrift.cpp:346] Thrift: TPipe ::GetOverlappedResult errored GLE=: errno = 109
I0214 19:32:11.105000  6068 impl_thrift.cpp:346] Thrift: TConnectedClient died: TPipe: GetOverlappedResult failed
I0214 19:32:11.134999  6060 impl_thrift.cpp:346] Thrift: TPipe ::GetOverlappedResult errored GLE=: errno = 109
I0214 19:32:11.136999  6060 impl_thrift.cpp:346] Thrift: TConnectedClient died: TPipe: GetOverlappedResult failed
```

I've placed it behind a flag so we can reduce the noise in verbose mode.

### CODE REVIEW

**Comment by theopolis** [comment]
I'd flip the logic and call this thirft_verbose and set it to false by default. Then add to the documentation where "verbose" shows up and indicate this flag can be turned on for more debugging.

**Comment by Breakwell** [comment]
> I'd flip the logic and call this thirft_verbose and set it to false by default. Then add to the documentation where "verbose" shows up and indicate this flag can be turned on for more debugging.

Done, is it just the docs/wiki/installation/cli-flags.md I need to update?

File: osquery/extensions/impl_thrift.cpp [hunk 1]
```diff
@@ -369,8 +371,10 @@ void ExtensionRunnerInterface::init(RouteUUID uuid, bool manager) {
         std::make_shared<extensions::ExtensionManagerProcessor>(handler);
   }
   // Set the global output function for thrift
-  GlobalOutput.setOutputFunction(
-      [](const char* message) -> void { VLOG(1) << "Thrift: " << message; });
+  if (FLAGS_thrift_verbose) {
+    GlobalOutput.setOutputFunction(
```
Comment (theopolis) [inline osquery/extensions/impl_thrift.cpp:375]: I am curious what happens when we do not call `setOutputFunction`? Is it the case that there is a default, thrift-provided, output function?

Do we need to set this to a no-op when our flag is not enabled?

Comment (Breakwell) [reply osquery/extensions/impl_thrift.cpp:375]: So the default is set [here](https://github.com/apache/thrift/blob/077b5fce825e79d84592fff893639b92b637eec7/lib/cpp/src/thrift/TOutput.cpp#L36) and ends up calling:
https://github.com/apache/thrift/blob/077b5fce825e79d84592fff893639b92b637eec7/lib/cpp/src/thrift/TOutput.cpp#L92

Which will cause it to print out to the stderr. I thought that the thrift [CmakeList.txt](https://github.com/osquery/osquery/blob/master/libraries/cmake/source/thrift/CMakeLists.txt) had defined ```THRIFT_SQUELCH_CONSOLE_OUTPUT``` at one point but maybe I've seen that somewhere else. Either we define it, accept the prints (which I think [this](https://github.com/osquery/osquery/pull/6592) was originally removing) I can't see a ``` if(!f_)``` so we'd probably have to do something like 
```
GlobalOutput.setOutputFunction(
      [](const char* message) -> void {});
```
to remove the call.

Comment (theopolis) [reply osquery/extensions/impl_thrift.cpp:375]: Ah cool, thanks for researching and confirming. Right, this change should add that no-op lambda you recommend if the flag is not enabled.

Comment (Breakwell) [reply osquery/extensions/impl_thrift.cpp:375]: Done

**Review by theopolis — APPROVED** [review]

**Pull request merged**